packages/qdrant-loader/src/qdrant_loader/connectors/jira/connector.py

```
@@ -446,27 +447,53 @@ def _get_issue_attachments(self, issue: JiraIssue) -> list[AttachmentMetadata]:
 
         return attachment_metadata
 
-    async def get_documents(self) -> list[Document]:
-        """Fetch and process documents from Jira.
+    async def fetch_by_id(self, entity_id: str) -> Document | None:
+        """Fetch a single Jira issue by key or numeric ID."""
+        try:
+            raw_issue = await self._make_request("GET", f"issue/{entity_id}")
+        except Exception as exc:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)